nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/ApiModel/Job.cs

```diff
@@ -109,6 +111,90 @@ public bool IsDependencyIgnored(string dependencyName, string dependencyVersion)
             .Any(r => r.IsSatisfiedBy(parsedDependencyVersion));
         return isIgnored;
     }
+
+    public bool IsUpdatePermitted(Dependency dependency)
+    {
+        if (dependency.Version is null)
+        {
+            // if we don't know the version, there's nothing we can do
+            return false;
+        }
+
+        var version = NuGetVersion.Parse(dependency.Version);
+        var dependencyInfo = RunWorker.GetDependencyInfo(this, dependency);
+        var isVulnerable = dependencyInfo.Vulnerabilities.Any(v => v.IsVulnerable(version));
+        var allowed = AllowedUpdates.Any(allowedUpdate =>
+        {
+            // check name restriction, if any
+            if (allowedUpdate.DependencyName is not null)
+            {
+                var matcher = new Matcher(StringComparison.OrdinalIgnoreCase)
+                    .AddInclude(allowedUpdate.DependencyName);
+                var result = matcher.Match(dependency.Name);
+                if (!result.HasMatches)
+                {
+                    return false;
+                }
+            }
+
+            var isSecurityUpdate = allowedUpdate.UpdateType == UpdateType.Security || SecurityUpdatesOnly;
```

Comment:
Should this be 
```suggestion
            var isSecurityUpdate = allowedUpdate.UpdateType == UpdateType.Security && SecurityUpdatesOnly;
```
instead?

Reply:
This is a direct port of the Ruby and `||` is used [there](https://github.com/dependabot/dependabot-core/blob/c9d3c7a592c654e8c046cbaa03f007ac17abd5ac/updater/lib/dependabot/job.rb#L276).